src/app/components/add-task-form/add-task-form.component.ts

```diff
@@ -37,6 +38,18 @@ export class AddTaskComponent implements OnInit {
       description: [""],
       type: ["", {validators: [Validators.required]}]
     })
+
+  }
+
+  ngOnDestroy(): void {
+    this.dialogRef.closeAll();
+  }
+
+  ngAfterViewInit(): void {
+    this.openForm();
+    this.dialogRef.afterAllClosed.subscribe(() => {
```

Comment:
why do you use ngAfterViewInit instead of just ngOnInit?

Reply:
Because ViewChild doesn't get executed before ngOnInit and the openForm function would try and open a dialog with an empty template

Reply:
There is no reason to make a modal if you're just navigating to a different page, if you want to use a modal, then you should open the modal when click the add-task-button component, which will open your modal, otherwise if you want to use a new page, there's no reason for it to be a modal. It just doesn't make sense to create a modal to display on a new page, that's not what it's used for

Reply:
Originally it opened on the button click, but the requirement was to create a route so I just moved the modal over to a new page, I'm going to remove the modal part and just keep it as a regular component then

Reply:
Sounds good to me 👍